### PR TITLE
dakota: fix build dependencies 

### PR DESCRIPTION
**Pull request by kean-smullen-nnl**
Dakota fails to install without perl-data-dumper on the system
`Can't locate Data/Dumper.pm in @INC (you may need to install the Data::Dumper module)`

### CODE REVIEW

File: var/spack/repos/builtin/packages/dakota/package.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ class Dakota(CMakePackage):
     depends_on('mpi', when='+mpi')
 
     depends_on('python')
+    depends_on('perl-data-dumper')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/dakota/package.py]: Is this a dependency for all versions or only those from a specific point on?

Also, if this is only a build dependency you could add `, type='build'`.

Comment (kean-smullen-nnl) [reply var/spack/repos/builtin/packages/dakota/package.py]: Updated for version 6.12 and above and added your suggested type='build'

**Comment by kean-smullen-nnl** [comment]
@tldahlgren Addressed your comment, please let me know if there's anything else to fix.

**Review by haampie — APPROVED** [review]

File: var/spack/repos/builtin/packages/dakota/package.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ class Dakota(CMakePackage):
     depends_on('mpi', when='+mpi')
 
     depends_on('python')
+    depends_on('perl-data-dumper', type='build', when='@:6.12')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/dakota/package.py]:
```suggestion
    depends_on('perl-data-dumper', type='build', when='@:6.12')
    depends_on('perl-data-dumper', type='build', when='@6.12:')
```

`@6.12:` is 6.12 and above, `@:6.12` is everything strictly < 6.13

**Review by haampie — CHANGES_REQUESTED** [review]

**Review by haampie — APPROVED** [review]
Thanks!

**Pull request merged**